cmd/influx/secret.go

```diff
@@ -0,0 +1,184 @@
+)
+
+type secretSVCsFn func() (influxdb.SecretService, influxdb.OrganizationService, func(*input.UI) string, error)
+
+func cmdSecret(opts ...genericCLIOptFn) *cobra.Command {
+	return newCmdSecretBuilder(newSecretSVCs, opts...).cmd()
+}
+
+type cmdSecretBuilder struct {
+	genericCLIOpts
+
+	svcFn secretSVCsFn
+
+	key string
+	org organization
+}
+
+func newCmdSecretBuilder(svcsFn secretSVCsFn, opts ...genericCLIOptFn) *cmdSecretBuilder {
+	opt := genericCLIOpts{
+		in: os.Stdin,
+		w:  os.Stdout,
+	}
+	for _, o := range opts {
+		o(&opt)
+	}
+
+	return &cmdSecretBuilder{
+		genericCLIOpts: opt,
+		svcFn:          svcsFn,
+	}
+}
+
+func (b *cmdSecretBuilder) cmd() *cobra.Command {
+	cmd := b.newCmd("secret", nil)
+	cmd.Short = "Secret management commands"
+	cmd.Run = seeHelp
+	cmd.AddCommand(
+		b.cmdDelete(),
+		b.cmdFind(),
+		b.cmdUpdate(),
+	)
+	return cmd
+}
+
+func (b *cmdSecretBuilder) cmdUpdate() *cobra.Command {
+	cmd := b.newCmd("update", b.cmdUpdateRunEFn)
+	cmd.Short = "Update secret"
+	cmd.Flags().StringVarP(&b.key, "key", "k", "", "The secret key (required)")
+	cmd.MarkFlagRequired("key")
+	b.org.register(cmd, false)
+
+	return cmd
+}
+
+func (b *cmdSecretBuilder) cmdDelete() *cobra.Command {
+	cmd := b.newCmd("delete", b.cmdDeleteRunEFn)
+	cmd.Short = "Delete secret"
+
+	cmd.Flags().StringVarP(&b.key, "key", "k", "", "The secret key (required)")
+	cmd.MarkFlagRequired("key")
+	b.org.register(cmd, false)
+
+	return cmd
+}
+
+func (b *cmdSecretBuilder) cmdUpdateRunEFn(cmd *cobra.Command, args []string) error {
+	scrSVC, orgSVC, getSecretFn, err := b.svcFn()
+	if err != nil {
+		return err
+	}
+	orgID, err := b.org.getID(orgSVC)
+	if err != nil {
+		return err
+	}
+
+	ctx := context.Background()
+
+	ui := &input.UI{
+		Writer: b.genericCLIOpts.w,
+		Reader: b.genericCLIOpts.in,
+	}
+	secret := getSecretFn(ui)
+
+	if err := scrSVC.PatchSecrets(ctx, orgID, map[string]string{b.key: secret}); err != nil {
+		return fmt.Errorf("failed to update secret with key %q: %v", b.key, err)
+	}
+
+	w := b.newTabWriter()
+	w.WriteHeaders("Key", "OrgID", "Updated")
+	w.Write(map[string]interface{}{
+		"Key":     b.key,
+		"OrgID":   orgID,
+		"Updated": true,
+	})
+	w.Flush()
+
+	return nil
+}
+
+func (b *cmdSecretBuilder) cmdDeleteRunEFn(cmd *cobra.Command, args []string) error {
+	scrSVC, orgSVC, _, err := b.svcFn()
+	if err != nil {
+		return err
+	}
+	orgID, err := b.org.getID(orgSVC)
+	if err != nil {
+		return err
+	}
+
+	ctx := context.Background()
+	if err := scrSVC.DeleteSecret(ctx, orgID, b.key); err != nil {
+		return fmt.Errorf("failed to delete secret with key %q: %v", b.key, err)
+	}
+
+	w := b.newTabWriter()
+	w.WriteHeaders("Key", "OrgID", "Deleted")
+	w.Write(map[string]interface{}{
+		"Key":     b.key,
+		"OrgID":   orgID,
+		"Deleted": true,
+	})
+	w.Flush()
+
+	return nil
+}
+
+func (b *cmdSecretBuilder) cmdFind() *cobra.Command {
+	cmd := b.newCmd("find", b.cmdFindRunEFn)
+	cmd.Short = "Find secrets"
+	b.org.register(cmd, false)
```

Comment:
This is an overall critique of "find"/"list" commands throughout the `influx` CLI. We are inconsistent. For some it's "list" and others it's "find". The term "find" indicates a searching action and I would expect ways to filter the results. This just lists all the secret keys.

My personal preference would be to make everything consistent and use either `list` or `ls` to align with standard Linux/UNIX command line tools.

@jademcgough

Reply:
@sanderson, this x1000! I've had the same feelings towards list|find. I would really love to see it become list if it does not allow for filtering. We do have filtering on some Find commands, either by name or what not. But these are still akin to list operations for me in most nix envs. I would love to see `list` and `ls` as an alias shorthand for calling it 👍

Reply:
@sanderson we're going to make these changes 👍. @influxdata/tools-team is on board 🤘

Reply:
@jsteenb2 🎉